src/commons/generic/include/pdc_generic.h

```diff
@@ -18,135 +19,141 @@ typedef enum { false = 0, true = 1 } bool;
 #endif
 
 typedef enum pdc_c_var_type_t {
-    PDC_UNKNOWN    = -1, /* error                                                          */
-    PDC_INT        = 0,  /* integer types     (identical to int32_t)                       */
-    PDC_FLOAT      = 1,  /* floating-point types                                           */
-    PDC_DOUBLE     = 2,  /* double types                                                   */
-    PDC_CHAR       = 3,  /* character types                                                */
-    PDC_STRING     = 4,  /* string types                                                   */
-    PDC_BOOLEAN    = 5,  /* boolean types                                                  */
-    PDC_SHORT      = 6,  /* short types                                                    */
-    PDC_UINT       = 7,  /* unsigned integer types (identical to uint32_t)                 */
-    PDC_INT64      = 8,  /* 64-bit integer types                                           */
-    PDC_UINT64     = 9,  /* 64-bit unsigned integer types                                  */
-    PDC_INT16      = 10, /* 16-bit integer types                                           */
-    PDC_INT8       = 11, /* 8-bit integer types                                            */
-    PDC_UINT8      = 12, /* 8-bit unsigned integer types                                   */
-    PDC_UINT16     = 13, /* 16-bit unsigned integer types                                  */
-    PDC_INT32      = 14, /* 32-bit integer types                                           */
-    PDC_UINT32     = 15, /* 32-bit unsigned integer types                                  */
-    PDC_LONG       = 16, /* long types                                                     */
-    PDC_VOID_PTR   = 17, /* void pointer type                                              */
-    PDC_SIZE_T     = 18, /* size_t type                                                    */
-    PDC_TYPE_COUNT = 19  /* this is the number of var types and has to be the last         */
+    PDC_UNKNOWN    = 0,  /* error */
+    PDC_SHORT      = 1,  /* short types */
```

Comment:
write a comment about the functions are depending on the ordering of the enum values.